src/main/java/com/puppycrawl/tools/checkstyle/XmlLoader.java

```diff
@@ -119,10 +119,10 @@ public void error(SAXParseException exception) throws SAXException {
         public static final String ENABLE_EXTERNAL_DTD_LOAD = "checkstyle.enableExternalDtdLoad";
 
         /** Feature that enables loading external DTD when loading XML files. */
-        private static final String LOAD_EXTERNAL_DTD =
+        public static final String LOAD_EXTERNAL_DTD =
                 "http://apache.org/xml/features/nonvalidating/load-external-dtd";
         /** Feature that enables including external general entities in XML files. */
-        private static final String EXTERNAL_GENERAL_ENTITIES =
+        public static final String EXTERNAL_GENERAL_ENTITIES =
```

Comment:
@romani I understand the other changes, but why do these fields need to be public?

Reply:
to reuse them at https://github.com/checkstyle/checkstyle/pull/6577/commits/6deec2134d7a66909a5290e0bb1ac6ca7ac06e98#diff-7771d8a02d03fa72c692d399a84b8f28R54

they are final and static, we use them in one Xml loader, so reuse them in another.

reuse of LoadExternalDtdFeatureProvider is not possible, as factories are of different type.

Reply:
Sorry I missed that.